theano/tensor/tests/test_opt.py

```diff
@@ -1811,6 +1811,18 @@ def test_local_useless_subtensor():
         f([[0, 1, 2], [3, 4, 5]])  # let debugmode test something
 
 
+class test_subtensor_inc_subtensor(unittest.TestCase):
+    def test1(self):
+        x = tensor.matrix('x')
+        i = tensor.iscalar('i')
+        v = tensor.vector('v')
+        y = tensor.set_subtensor(x[i], v)
+        z = y[i]
+        f = theano.function([x, i, v], z)
+        prog = f.maker.fgraph.toposort()
+        assert len(prog) == 1
```

Comment:
execute the function and execute it and check the result, just to be sure it give the right answer.